fake.js

```diff
@@ -53,4 +53,38 @@ module.exports = class FakeTransaction extends Transaction {
       this.from = data.from
     }
   }
+
+  /**
+   * Computes a sha3-256 hash of the serialized tx
+   * @param {Boolean} [includeSignature=true] whether or not to inculde the signature
+   * @return {Buffer}
+   */
+  hash (includeSignature) {
+    if (includeSignature === undefined) includeSignature = true
+
+    // EIP155 spec:
+    // when computing the hash of a transaction for purposes of signing or recovering,
+    // instead of hashing only the first six elements (ie. nonce, gasprice, startgas, to, value, data),
+    // hash nine elements, with v replaced by CHAIN_ID, r = 0 and s = 0
+
+    let items
+    if (includeSignature) {
+      items = this.raw
+      items += this.getSenderAddress();
```

Comment:
I think instead of duplicating, `fakeTransaction` should change `.raw` in the constructor to include the signature (a fake one perhaps). The `hash` method then is inherited unchanged from the normal implementation.

This is a lot of code duplication otherwise.

Reply:
@axic I think I can accommodate that easily enough. However, does that mean that instead of overriding `hash` we'll need to override `sign`?

Reply:
Also thanks for the fast response!

Reply:
@axic see latest changes. I'm not sure that I like this however, as I feel `sign` should probably be overridden as well to avoid cases where the fake signature is overwritten.